sdk/communication/Azure.Communication.Rooms/src/Models/CommunicationIdentifierSerializer.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Text.Json;
+
+namespace Azure.Communication.Rooms.Models
+{
+    // The CommunicationIdentifierSerializer is a clone of Shared/CommunicationIdentifireSerializer.cs file.
+    // Right now Rooms do not support Phone number and Team's User Identifier.
+    // Once Rooms support all identifiers as defined by Shared/CommunicationIdentifireSerializer.cs we will refer to the Shard class.
```

Comment:
I believe it's safer to use the shared class, otherwise if everyone of our SDK copies we end up with diverging code in a piece that is critical. Let's please use the shared Serializer.

Reply:
@DominikMe thanks for taking the time to look into the PR.
Right now, Rooms only support Communication User Identifier. When defining the rooms api [spec](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/communication/data-plane/Rooms/preview/2022-02-01-preview/communicationservicesrooms.json#L510), had to define a custom model for Communication Identifier Model in-order to exclude phone number and team's user. 
That translated into having to define custom serializer to avoid build issues as phone number and teams' user are not present in the model. 

Open to recommendation in such scenario, where teams don't support all identifiers from api spec and sdk perspective.
[Can discuss further on the discussion chat and incorporate accordingly on upcoming refresh]

Reply:
Further discussions are on chat thread, will be addressed in upcoming revisions.

Reply:
Let's keep unresolved and discuss

Reply:
Tracked internally under [User Story 2946996](https://skype.visualstudio.com/SPOOL/_workitems/edit/2946996): _[Rooms] SDK should use shared identifier serializer_ for next preview version.